### PR TITLE
feat: Adds variable to use custom config file with Elixir Credo

### PR DESCRIPTION
**Pull request by Thau**
Adds `g:ale_elixir_credo_config_file` to allow users to use a custom configuration file for the checks. Motivation is my company using a centralised configuration file for all of our Elixir projects, so it's our belief that more people could benefit from this option.

Apart from that, this PR also adds tests in general for Credo calls, because there were none before.

Thank you!

### CODE REVIEW

**Comment by Thau** [comment]
Seems I'm causing some test errors when using Windows. Checking how to fix those.

**Comment by Thau** [comment]
There's 50 open PRs, so I won't be bothering anyone to merge this any time soon. Just ping me when you want, and I'll rebase it!

**Comment by Thau** [comment]
Was this automatically because there was no movement, or was it just rejected?

**Comment by Thau** [comment]
Excuse me, @w0rp, could we have a look at reopening this one, please?

**Comment by w0rp** [comment]
GitHub won't let me reopen this pull request due to a force push on the branch, but you can open another one.

The one change you need to make is to move your tests into the file for the existing suite of tests: `test/command_callback/test_elixir_credo.vader`.

I'll leave a suggestion for a simplification you can make.

File: ale_linters/elixir/credo.vim [hunk 1]
```diff
@@ -45,13 +45,24 @@ function! ale_linters#elixir#credo#GetMode() abort
...
 endfunction
 
+function! ale_linters#elixir#credo#GetConfigFile() abort
+    let l:config_file = get(g:, 'ale_elixir_credo_config_file', '')
+
+    if len(l:config_file) == 0
+        return ''
+    else
+        return ' --config-file ' . l:config_file
+    endif
```
Comment (w0rp) [inline ale_linters/elixir/credo.vim:55]: GitHub won't let me post a suggestion to a closed pull request, but you can use `empty()` to check if the config value is empty, and you can remove the redundant `else` and just `return` at the end of the function.

Comment (Thau) [reply ale_linters/elixir/credo.vim:55]: Hey, thanks! I’ll improve it and reopen a new PR. Thanks for the answer and sorry for pinging you directly!
